source/der.c

```diff
@@ -98,6 +102,10 @@ static int s_der_read_tlv(struct aws_byte_cursor *cur, struct der_tlv *tlv) {
         len = len_bytes;
     }
 
+    if (len > cur->len) {
+        return aws_raise_error(AWS_ERROR_CAL_MALFORMED_ASN1_ENCOUNTERED);
+    }
```

Comment:
Is it OK that this function incremented the cursor?
Are there any assumptions in the calling code that the cursor is unchanged if the function fails?

If this error results in the calling code just erroring out, then shipit.

If the calling code tries to do something with the cursor after an error, then NO-shipit

Reply:
yeah, this function is invoked when parsing der during decoder_new. any error in decoding just gets propagated back up and we return NULL decoder back. failing to read length field also results in that flow.